e2e/mock-api-v2/src/helpers/match.ts

```diff
@@ -0,0 +1,31 @@
+import { Schema } from '@effect/schema';
+import { Effect, Match } from 'effect';
+import { InvalidUsernamePassword, InvalidProtectNode } from '../errors';
+import { PingOneCustomHtmlRequestBody } from '../schemas/customHtmlTemplate/requests';
+
+type PingRequestData = Schema.Schema.Type<
+  typeof PingOneCustomHtmlRequestBody
+>['parameters']['data']['formData']['value'];
+/**
+ * Using this to match on the data types, realistically, this will be a schema of possible
+ * response bodies we want to validate against they validate to our conditions.
+ *
+ * We can then return back either an Error to respond with, if validation fails
+ * or we can continue to the next step in the flow
+ */
+const validator = Match.type<PingRequestData>().pipe(
+  Match.when({ username: Match.string, password: Match.string }, ({ username, password }) => {
+    return Effect.if(username == 'testuser' && password === 'Password', {
+      onFalse: () => Effect.fail(new InvalidUsernamePassword()),
+      onTrue: () => Effect.succeed(true),
+    });
+  }),
+  Match.when({ pingprotectsdk: Match.string }, ({ pingprotectsdk }) =>
+    Effect.if(pingprotectsdk.length > 1, {
```

Comment:
are you saying are we missing a `return` on the `Match.when`?

its actually part of the `pipe`. 

a Match has a few pieces. the initial piece of it is, what are we matching on, in this case its the `type` passed as a generic.

Then we `pipe` that code and have `when` conditions, that result in some function applied `when` true.

I used `Effect.if` to handle the validation logic as it was just to get it done, there may be better ways to do this.

If I misunderstood, we have an implicit `return` on the arrow function. I can break it out to be an explicit `return` if thats better to read.

Reply:
I'm just asking because you have a `return` on line 18 with the `Effect.if` statement, but you don't have it on line 24 with that `Effect.if`. Attached is a screenshot. I doubt this inconsistency is consequential, but I brought it up just in case it was.

![CleanShot 2024-07-12 at 17 53 51](https://github.com/user-attachments/assets/310a579d-43da-4ee0-ad0a-65e7002984d8)

Reply:
Yeah line 24 just has an implicit return. I can standardize it